src/main/java/org/mockbukkit/mockbukkit/inventory/PlayerInventoryMock.java

```diff
@@ -77,8 +77,25 @@ public HumanEntity getHolder()
 	@Override
 	public void setStorageContents(ItemStack[] items) throws IllegalArgumentException
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException("setStorageContests has not been implemented for Player Inventories");
+		if (items == null)
+		{
+			throw new NullPointerException("ItemStack was null");
+		}
+		else if (items.length > 36)
+		{
+			throw new IllegalArgumentException("ItemStack array too large (max: 36, was: " + items.length + ")");
+		}
+		for (int i = 0; i < getSize(); i++)
+		{
+			if (i < items.length && items[i] != null)
+			{
+				setItem(i, items[i]);
+			}
+			else
+			{
+				setItem(i, null);
+			}
+		}
```

Comment:
I noticed that a similar logic already exists in InventoryMock.
https://github.com/MockBukkit/MockBukkit/blob/4bd5c3396731cf0bc112a23523b40f054f336e53/src/main/java/org/mockbukkit/mockbukkit/inventory/InventoryMock.java#L328-L342
If we remove this method definition is it enough to fix it? If so, we can avoid duplicating the logic :)

Reply:
From my point of view, replicating a bit of code is fine to do here and there. 

This implementation looks like it clears all armor contents, as it iterates through the size of the whole inventory, is that intended behavior? (I don't have the time right now to compare with paper code or test this)